Tests/UnitTests/Core/Propagator/MaterialCollectionTests.cpp

```diff
@@ -101,7 +101,20 @@ void runTest(const propagator_t& prop, double pT, double phi, double theta,
   double q = dcharge;
   Vector3D pos(x, y, z);
   Vector3D mom(px, py, pz);
-  CurvilinearParameters start(std::nullopt, pos, mom, q, time);
+
+  BoundSymMatrix cov;
+  // take some major correlations (off-diagonals)
+  // clang-format off
+    cov <<
+     10_mm, 0, 0.123, 0, 0.5, 0,
+     0, 10_mm, 0, 0.162, 0, 0,
+     0.123, 0, 0.1, 0, 0, 0,
+     0, 0.162, 0, 0.1, 0, 0,
+     0.5, 0, 0, 0, 1_e / 10_GeV, 0,
+     0, 0, 0, 0, 0, 1_us;
+  // clang-format on
+  std::cout << cov.determinant() << std::endl;
+  CurvilinearParameters start(cov, pos, mom, q, time);
```

Comment:
Why is the `MaterialCollection` now done with a covariance matrix?

Reply:
This was one of the things I added since the code coverage was too low. All branches inside the `MaterialInteractor` that involve the covariance matrix were not tested at all. In order to increase the coverage I added a brief test to see that the matrix will change the right way due to material interaction.